database.go

```diff
@@ -140,6 +140,22 @@ func (db *Database) LockProtectedEntries() error {
 	return nil
 }
 
+// AddBinary adds a binary to the database.
+// It takes care of adding it to the correct place based on the format version
+func (db *Database) AddBinary(binaryContent []byte) *Binary {
+	if db.Header.IsKdbx4() {
```

Comment:
I think we can add
```
func (db *Database) GetBinaries() *Binaries {
	if db.Header.IsKdbx4() {
		return &(db.Content.InnerHeader.Binaries)
	}
	return &(db.Content.Meta.Binarie)
}
```

And use it here and in func (db *Database) FindBinary

Reply:
Do we need a pointer to binaries here? It is just an alias for a slice of `Binary`

Reply:
Can we modify the database binaries if return just a slice? For example I need to remove one of the binaries item.

Reply:
Generally speaking, yes, that should work.
I'd say `RemoveBinary(id)` would be more consistent towards us moving functions onto the DB level.

If we keep exposing `Binaries` directly for necessary usage, then it will be used and people might be using `Add` because the `Binaries` are already there

Reply:
I added a `RemoveBinary(id)` function which takes care of modifying the database.

Without breaking changes on the structure this wouldn't really work though